docs/importing-js.md

```diff
@@ -12,21 +12,36 @@ Most components ship with Component / Foundation classes which are used to provi
 
 ## ES2015
 
-```javascript
+```js
 import {MDCFoo, MDCFooFoundation} from '@material/foo';
 ```
 
+Note that MDC Web's packages point `main` to pre-compiled UMD modules under `dist` to maximize compatibility.
+Build toolchains often assume that dependencies under `node_modules` are already ES5, and thus skip transpiling them.
+
+However, if you want to take advantage of tree-shaking and dependency sharing within MDC Web's code to reduce the size
+of your built assets, you will want to explicitly reference the package's `index.js`, which contains the ES2015+ source:
+
+```js
+import {MDCFoo, MDCFooFoundation} from '@material/foo/index';
+```
+
+Note that in this case, you must ensure your build toolchain is configured to process and transpile MDC Web's modules
+as well as your own. You will also need to include babel's `transform-object-assign` plugin for IE 11 support.
```

Comment:
This piece of info should also be in the getting-started.md

https://github.com/material-components/material-components-web/blame/master/docs/getting-started.md#L222

Reply:
Yeah, I had intended to look into the right place to add that to getting-started, thanks for reminding me.

Reply:
I've fixed that in #3665, since that PR was already touching the lines around that point in the file. That way hopefully it'll merge cleanly the next time we sync this PR.